<a href="https://colab.research.google.com/github/Younoone/Practice/blob/main/EDA_ASSIGNMENT.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#EDA

Hi everyone, This is your first assignment for week 1...
So in this assignment you have to perform EDA on the below dataset.
The problem statements are given below.
Just write a code for them...

In [ ]:
!wget https://www.kaggle.com/datasets/aestheteaman01/zomato-restaurants-in-delhi-ncr


--2023-09-09 13:43:50--  https://www.kaggle.com/datasets/aestheteaman01/zomato-restaurants-in-delhi-ncr
Resolving www.kaggle.com (www.kaggle.com)... 35.244.233.98
Connecting to www.kaggle.com (www.kaggle.com)|35.244.233.98|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘zomato-restaurants-in-delhi-ncr’

zomato-restaurants-     [ <=>                ]  15.94K  --.-KB/s    in 0.001s  

2023-09-09 13:43:50 (23.8 MB/s) - ‘zomato-restaurants-in-delhi-ncr’ saved [16322]



Q1.Are there any null values in the dataset?if yes then fill them..

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

In [2]:
Zomato = pd.read_csv("DelhiNCR Restaurants.csv")
Zomato

Restaurant_Name  \
0                   Rustom's   
1                  Cafe Lota   
2     Dum-Pukht - ITC Maurya   
3                Burma Burma   
4              The Big Chill   
...                      ...   
1960          Bablu Chic-Inn   
1961               Kovilakam   
1962        Krips Restaurant   
1963                Cafe 147   
1964                Pizzasia   

                                               Category  Pricing_for_2  \
0     Parsi, Street Food, North Indian, Desserts, Be...           2100   
1           Cafe, South Indian, North Indian, Beverages           1200   
2                       Mughlai, North Indian, Desserts           5000   
3           Asian, Burmese, Bubble Tea, Desserts, Salad           1600   
4                        Continental, American, Italian           1500   
...                                                 ...            ...   
1960          Mughlai, North Indian, Chinese, Fast Food            500   
1961                                       South Indian            350   
1962                     North Indian, Chinese, Mughlai            900   
1963  Cafe, North Indian, Chinese, Kashmiri, Rolls, ...            800   
1964                        Pizza, Fast Food, Beverages            500   

                                         Locality  Dining_Rating  \
0                                  ITO, New Delhi            4.9   
1                       Pragati Maidan, New Delhi            4.9   
2             ITC Maurya, Chanakyapuri, New Delhi            4.9   
3              Cyber Hub, DLF Cyber City, Gurgaon            4.9   
4                          Khan Market, New Delhi            4.9   
...                                           ...            ...   
1960               Ashok Vihar Phase 3, New Delhi            3.9   
1961                    Dilshad Garden, New Delhi            3.9   
1962   Basant Lok Market, Vasant Vihar, New Delhi            3.9   
1963                      Sainik Farms, New Delhi            3.9   
1964                   Gaur City 2, Greater Noida            3.9   

      Dining_Review_Count  Delivery_Rating  Delivery_Rating_Count  \
0                    1885              4.4                   1844   
1                    3748              3.9                     37   
2                    1371              NaN                      0   
3                    2636              4.4                   1238   
4                    6487              NaN                      0   
...                   ...              ...                    ...   
1960                  183              4.1                    545   
1961                  114              NaN                      1   
1962                  316              3.9                   2977   
1963                  120              3.8                    196   
1964                  165              3.9                   5196   

                                                Website  \
0      https://www.zomato.com/ncr/rustoms-ito-new-delhi   
1     https://www.zomato.com/ncr/cafe-lota-pragati-m...   
2     https://www.zomato.com/ncr/dum-pukht-itc-maury...   
3     https://www.zomato.com/ncr/burma-burma-dlf-cyb...   
4     https://www.zomato.com/ncr/the-big-chill-khan-...   
...                                                 ...   
1960  https://www.zomato.com/ncr/bablu-chic-inn-asho...   
1961  https://www.zomato.com/ncr/kovilakam-dilshad-g...   
1962  https://www.zomato.com/ncr/krips-restaurant-va...   
1963  https://www.zomato.com/ncr/cafe-147-sainik-far...   
1964  https://www.zomato.com/ncr/pizzasia-gaur-city-...   

                                                Address     Phone_No  \
0     Delhi Parsi Anjuman, LNJP Colony, Bahadur Shah...   9.1991E+11   
1     National Crafts Museum, Gate 2, Bhairon Marg, ...  9.17839E+11   
2                   ITC Maurya, Chanakyapuri, New Delhi  9.11146E+11   
3     Shop 6, Ground Floor, Building 8, Tower C, Cyb...  9.11244E+11   
4                       

In [3]:
Zomato.isna().sum()

Restaurant_Name            0
Category                   0
Pricing_for_2              0
Locality                   0
Dining_Rating              0
Dining_Review_Count        0
Delivery_Rating          402
Delivery_Rating_Count      0
Website                    0
Address                    0
Phone_No                   0
Latitude                   0
Longitude                  0
Known_For2               405
Known_For22              841
dtype: int64

In [4]:
Zomato["Delivery_Rating"].unique()

array([4.4, 3.9, nan, 4.2, 3.8, 4.5, 4. , 4.1, 4.3, 3.2, 3.7, 3.4, 3.5,
       3.6, 3.1, 2.8, 3.3, 2.9, 3. ])

In [5]:
mean = Zomato["Delivery_Rating"].mean()
mean

3.964299424184261

In [6]:
Zomato["Delivery_Rating"]=Zomato["Delivery_Rating"].fillna(mean)

In [7]:
Zomato.isna().sum()

Restaurant_Name            0
Category                   0
Pricing_for_2              0
Locality                   0
Dining_Rating              0
Dining_Review_Count        0
Delivery_Rating            0
Delivery_Rating_Count      0
Website                    0
Address                    0
Phone_No                   0
Latitude                   0
Longitude                  0
Known_For2               405
Known_For22              841
dtype: int64

In [8]:
Zomato["Known_For2"].unique()

array([nan,
       'Pondicherry Fish Curry, Coconut Rabdi, Sabudana Pop Corn, Chicken Pasanda, Majestic Chicken, Bhapa Doi',
       'Royal ambiance and Mughlai cuisine, also received the "Times Food Award 2015" for Culinary Heritage',
       ...,
       'Excellent Food, Good Ambience, Courteous Staff, Awesome Food, Great Food, Ambiance',
       'Pool, Live Music, Cafe, Environment, Good Place, Staff',
       'Cordial Staff, Great Recommendations, Mouth Watering Food, Reasonable Prices, Quantity, Ambience'],
      dtype=object)

In [9]:
Zomato["Known_For2"].value_counts()

Get a complimentary night, earn up to 50% of spends as points at 250+ restaurants & much more. Know more    15
Finger Food, Continental, Mediterranean, Italian, Asian                                                      9
Continental, Italian, Fast Food, North Indian, Beverages, Coffee                                             7
Chinese, Asian, Continental, North Indian, Beverages                                                         2
Butterscotch Pastry, Truffle Pastry, Chocolate Truffle Cake, Pastries, Waffle, Cakes                         2
                                                                                                            ..
Passion Fruit Ice Tea, Cardamom Tea, Bun Tikki, Earl Grey Tea, Rose Tea, Smoked Chicken Pasta                1
Chicken Crostini, Blackberry Mojito, Butter Maggi, Peri Fries, Masala Tea, Hot Coffee                        1
Chicken Thai Curry, Iced Lemon Tea, Pizza Alle Verdure, Veggie Delight, Stroganoff, Chicken Shawarama        1
P

In [10]:
Zomato["Known_For2"]=Zomato["Known_For2"].fillna("Get a complimentary night, earn up to 50% of spends as points at 250+ restaurants & much more. Know more")

In [11]:
Zomato.isna().sum()

Restaurant_Name            0
Category                   0
Pricing_for_2              0
Locality                   0
Dining_Rating              0
Dining_Review_Count        0
Delivery_Rating            0
Delivery_Rating_Count      0
Website                    0
Address                    0
Phone_No                   0
Latitude                   0
Longitude                  0
Known_For2                 0
Known_For22              841
dtype: int64

In [12]:
Zomato["Known_For22"].unique()

array([nan,
       'Artistic Decor, The Service, Natural Ambience, Fusion Menu, Serene Ambience, Fusion Food',
       'Knowledgeable Staff, Authentic, Soothing Ambience, Dessert Menu, Pure Veg, Cozy',
       ...,
       'Casual Seating Area, Chinese Restaurant, Nice Crowd, Great Recommendations, Polite Staff, Perfect Presentation',
       'Live Entertainment, Good Bartenders, Beautiful Interior, Recommend Place, Dj, Cool Place',
       'Lip Smacking Food, Amazing Food, Owner, Good Place, Good Food, Quality'],
      dtype=object)

In [13]:
Zomato["Known_For22"].value_counts()

Pretty Decor, Perfect Presentation, Tamper Proof Packaging, Worth the Money, Packing, Food Quality                     2
Artistic Decor, The Service, Natural Ambience, Fusion Menu, Serene Ambience, Fusion Food                               1
Excellent Place, Great Value, Nice Service, Recommended Place, Good Quality Food, Amazing Taste                        1
Amazing Food and Experience, Delivery Instructions, Genuine Price, Faster Delivery, Worth the Price, Food Packaging    1
Budget Food, Good for Large Groups, Packed on Weekends, Portions, Good Taste, Main Course                              1
                                                                                                                      ..
Classy Decor, Wonderful Service, Chinese, Server, Excellent Food, Hospitality                                          1
Pocket Friendly, Nice Crowd, Great Ambiance, Polite Staff, Prompt Service, Worth the Money                             1
Happy Hour Menu, Balcony Area, A

In [14]:
Zomato["Known_For22"]=Zomato["Known_For22"].fillna("Pretty Decor, Perfect Presentation, Tamper Proof Packaging, Worth the Money, Packing, Food Quality")

In [15]:
Zomato.isna().sum()

Restaurant_Name          0
Category                 0
Pricing_for_2            0
Locality                 0
Dining_Rating            0
Dining_Review_Count      0
Delivery_Rating          0
Delivery_Rating_Count    0
Website                  0
Address                  0
Phone_No                 0
Latitude                 0
Longitude                0
Known_For2               0
Known_For22              0
dtype: int64

Q2.Which restaurant is the best one having ratings more than 4.5 for delivery


In [40]:
DRC=Zomato["Delivery_Rating"]
DRC

0       4.400000
1       3.900000
2       3.964299
3       4.400000
4       3.964299
          ...   
1960    4.100000
1961    3.964299
1962    3.900000
1963    3.800000
1964    3.900000
Name: Delivery_Rating, Length: 1965, dtype: float64

In [17]:
DRCC=DRC>4.5
DRCC

0       False
1       False
2       False
3       False
4       False
        ...  
1960    False
1961    False
1962    False
1963    False
1964    False
Name: Delivery_Rating, Length: 1965, dtype: bool

In [18]:
DRCC=DRCC.astype('int')
DRCC

0       0
1       0
2       0
3       0
4       0
       ..
1960    0
1961    0
1962    0
1963    0
1964    0
Name: Delivery_Rating, Length: 1965, dtype: int64

In [19]:
DRCC.sum()

0

In [20]:
DRCC.mean()

0.0

**There are no restaurants with delivery rating more than 4.5**


Q3.What type of food is most preffered by the customers?

In [ ]:
Zomato["Category"].value_counts()

In [ ]:
sns.countplot(x=Zomato["Category"])

In [ ]:
plt.figure(figsize=(15,10))
plt.hist(Zomato["Category"],bins=100)
plt.xlabel("Category")
plt.ylabel("Frequency")
plt.show()

In [ ]:
plt.figure(figsize=(120,10))
Zomato["Category"].value_counts().plot(kind='bar')

**"North Indian" is the most preferred food category**.

Q4.How many people prefer delivery rather than dining ?

In [21]:
DRC=Zomato['Delivery_Rating_Count']

In [22]:
DiRC=Zomato['Dining_Review_Count']

In [23]:
Arr = DRC-DiRC
Arr

0        -41
1      -3711
2      -1371
3      -1398
4      -6487
        ... 
1960     362
1961    -113
1962    2661
1963      76
1964    5031
Length: 1965, dtype: int64

In [24]:
Arr.sum()

4534427

**"4534427" no. of people prefer delivery over dining.**

Q5.Which restaurant is the best one having ratings more than 4.5 for dining?

In [25]:
Zomato

Restaurant_Name  \
0                   Rustom's   
1                  Cafe Lota   
2     Dum-Pukht - ITC Maurya   
3                Burma Burma   
4              The Big Chill   
...                      ...   
1960          Bablu Chic-Inn   
1961               Kovilakam   
1962        Krips Restaurant   
1963                Cafe 147   
1964                Pizzasia   

                                               Category  Pricing_for_2  \
0     Parsi, Street Food, North Indian, Desserts, Be...           2100   
1           Cafe, South Indian, North Indian, Beverages           1200   
2                       Mughlai, North Indian, Desserts           5000   
3           Asian, Burmese, Bubble Tea, Desserts, Salad           1600   
4                        Continental, American, Italian           1500   
...                                                 ...            ...   
1960          Mughlai, North Indian, Chinese, Fast Food            500   
1961                                       South Indian            350   
1962                     North Indian, Chinese, Mughlai            900   
1963  Cafe, North Indian, Chinese, Kashmiri, Rolls, ...            800   
1964                        Pizza, Fast Food, Beverages            500   

                                         Locality  Dining_Rating  \
0                                  ITO, New Delhi            4.9   
1                       Pragati Maidan, New Delhi            4.9   
2             ITC Maurya, Chanakyapuri, New Delhi            4.9   
3              Cyber Hub, DLF Cyber City, Gurgaon            4.9   
4                          Khan Market, New Delhi            4.9   
...                                           ...            ...   
1960               Ashok Vihar Phase 3, New Delhi            3.9   
1961                    Dilshad Garden, New Delhi            3.9   
1962   Basant Lok Market, Vasant Vihar, New Delhi            3.9   
1963                      Sainik Farms, New Delhi            3.9   
1964                   Gaur City 2, Greater Noida            3.9   

      Dining_Review_Count  Delivery_Rating  Delivery_Rating_Count  \
0                    1885         4.400000                   1844   
1                    3748         3.900000                     37   
2                    1371         3.964299                      0   
3                    2636         4.400000                   1238   
4                    6487         3.964299                      0   
...                   ...              ...                    ...   
1960                  183         4.100000                    545   
1961                  114         3.964299                      1   
1962                  316         3.900000                   2977   
1963                  120         3.800000                    196   
1964                  165         3.900000                   5196   

                                                Website  \
0      https://www.zomato.com/ncr/rustoms-ito-new-delhi   
1     https://www.zomato.com/ncr/cafe-lota-pragati-m...   
2     https://www.zomato.com/ncr/dum-pukht-itc-maury...   
3     https://www.zomato.com/ncr/burma-burma-dlf-cyb...   
4     https://www.zomato.com/ncr/the-big-chill-khan-...   
...                                                 ...   
1960  https://www.zomato.com/ncr/bablu-chic-inn-asho...   
1961  https://www.zomato.com/ncr/kovilakam-dilshad-g...   
1962  https://www.zomato.com/ncr/krips-restaurant-va...   
1963  https://www.zomato.com/ncr/cafe-147-sainik-far...   
1964  https://www.zomato.com/ncr/pizzasia-gaur-city-...   

                                                Address     Phone_No  \
0     Delhi Parsi Anjuman, LNJP Colony, Bahadur Shah...   9.1991E+11   
1     National Crafts Museum, Gate 2, Bhairon Marg, ...  9.17839E+11   
2                   ITC Maurya, Chanakyapuri, New Delhi  9.11146E+11   
3     Shop 6, Ground Floor, Building 8, Tower C, Cyb...  9.11244E+11   
4                       

In [28]:
DC=Zomato["Dining_Rating"]
max_rating = DC.max()
print(max_rating)

4.9


In [32]:
zomato_max_rating = Zomato[Zomato['Dining_Rating'] == max_rating]
zomato_max_rating.head()

Restaurant_Name                                           Category  \
0                Rustom's  Parsi, Street Food, North Indian, Desserts, Be...   
1               Cafe Lota        Cafe, South Indian, North Indian, Beverages   
2  Dum-Pukht - ITC Maurya                    Mughlai, North Indian, Desserts   
3             Burma Burma        Asian, Burmese, Bubble Tea, Desserts, Salad   
4           The Big Chill                     Continental, American, Italian   

   Pricing_for_2                              Locality  Dining_Rating  \
0           2100                        ITO, New Delhi            4.9   
1           1200             Pragati Maidan, New Delhi            4.9   
2           5000   ITC Maurya, Chanakyapuri, New Delhi            4.9   
3           1600    Cyber Hub, DLF Cyber City, Gurgaon            4.9   
4           1500                Khan Market, New Delhi            4.9   

   Dining_Review_Count  Delivery_Rating  Delivery_Rating_Count  \
0                 1885         4.400000                   1844   
1                 3748         3.900000                     37   
2                 1371         3.964299                      0   
3                 2636         4.400000                   1238   
4                 6487         3.964299                      0   

                                             Website  \
0   https://www.zomato.com/ncr/rustoms-ito-new-delhi   
1  https://www.zomato.com/ncr/cafe-lota-pragati-m...   
2  https://www.zomato.com/ncr/dum-pukht-itc-maury...   
3  https://www.zomato.com/ncr/burma-burma-dlf-cyb...   
4  https://www.zomato.com/ncr/the-big-chill-khan-...   

                                             Address     Phone_No   Latitude  \
0  Delhi Parsi Anjuman, LNJP Colony, Bahadur Shah...   9.1991E+11  28.638167   
1  National Crafts Museum, Gate 2, Bhairon Marg, ...  9.17839E+11  28.613429   
2                ITC Maurya, Chanakyapuri, New Delhi  9.11146E+11  28.598200   
3  Shop 6, Ground Floor, Building 8, Tower C, Cyb...  9.11244E+11  28.494464   
4                       68-A, Khan Market, New Delhi  9.11142E+11  28.600350   

   Longitude                                         Known_For2  \
0  77.240472  Get a complimentary night, earn up to 50% of s...   
1  77.242471  Pondicherry Fish Curry, Coconut Rabdi, Sabudan...   
2  77.173661  Royal ambiance and Mughlai cuisine, also recei...   
3  77.088532  Coconut Milk Dessert, Honey Caviar, Kaffir Lim...   
4  77.227493  Irish Cream Tiramisu, Penne Vodka, Mississippi...   

                                         Known_For22  
0  Pretty Decor, Perfect Presentation, Tamper Pro...  
1  Artistic Decor, The Service, Natural Ambience,...  
2  Pretty Decor, Perfect Presentation, Tamper Pro...  
3  Knowledgeable Staff, Authentic, Soothing Ambie...  
4  Retro Ambience, Yummy Desserts, Big Portions, ...

In [34]:
zomato_max_dining_count = zomato_max_rating["Dining_Review_Count"].max()
print(zomato_max_dining_count)

6487


In [38]:
best_rating_restaurant = zomato_max_rating[zomato_max_rating["Dining_Review_Count"]==zomato_max_dining_count]
print(best_rating_restaurant["Restaurant_Name"])

4    The Big Chill
Name: Restaurant_Name, dtype: object


**"The Big Chill" is the best restaurant having rating more than 4.5.**

Q6.Compare different delivery rating counts of different restaurants.
Also tell that which restaurants has got the highest rating counts.
Do this by creating a barplot

In [ ]:
Zomato

In [ ]:
Restaurants=Zomato["Restaurant_Name"]
print(Restaurants)
print(len(Restaurants))

In [ ]:
DRC=Zomato["Delivery_Rating_Count"]
DRC

In [ ]:
DRC.max()

In [ ]:
print(next(i for i,x in
          enumerate(DRC) if x==87700))

In [ ]:
DRC[136]

In [ ]:
Restaurants[136]

In [ ]:
plt.bar(Restaurants,DRC,width=0.5)
plt.xlabel("Restaurants")
plt.ylabel("DRC")
plt.show()

**"PUNJABI ANGITHI" has got the highest no. of rating counts.**